assets/js/components/ChecksCatalog/ChecksCatalogNew.jsx

```diff
@@ -1,66 +1,81 @@
 /* eslint-disable react/no-array-index-key */
-import React, { useEffect } from 'react';
+import React, { useState, useEffect } from 'react';
 import { useSelector, useDispatch } from 'react-redux';
 
 import { groupBy } from '@lib/lists';
 
 import { getCatalog } from '@state/selectors/catalog';
 import { updateCatalog } from '@state/actions/catalog';
+import { providersList } from '../ProviderLabel/ProviderLabel';
 import CatalogContainer from './CatalogContainer';
 import CheckItem from './CheckItem';
+import ProviderSelection from './ProviderSelection';
 
+const providers = ['default'].concat(providersList.slice());
 // eslint-disable-next-line import/prefer-default-export
 export function ChecksCatalogNew() {
   const dispatch = useDispatch();
-
+  const [selected, setSelected] = useState(providers[0]);
+  const provider = selected;
   const {
     data: catalogData,
     error: catalogError,
     loading,
-  } = useSelector(getCatalog());
+  } = useSelector(getCatalog(provider));
 
   useEffect(() => {
-    dispatch(updateCatalog());
-  }, [dispatch]);
+    setSelected(providers[0]);
+  }, [providers[0]]);
 
   useEffect(() => {
-    getCatalog();
-  }, []);
+    if (provider === 'default') {
+      dispatch(updateCatalog());
```

Comment:
I don't think we need this distinction.
If you set provider to `null`, it makes the same work